src/main/java/com/browserstack/automate/ci/common/tracking/PluginsTracker.java

```diff
@@ -27,14 +27,42 @@ public PluginsTracker(final String username, final String accessKey) {
         this.username = username;
         this.accessKey = accessKey;
         this.trackingId = Tools.getUniqueString(true, true);
+        initializeClient();
     }
 
     public PluginsTracker() {
         this.username = null;
         this.accessKey = null;
         this.trackingId = Tools.getUniqueString(true, true);
+        initializeClient();
     }
-
+    
+    public void initializeClient(){
+
+        String username = JenkinsProxySettings.getUsername();
+        String password = JenkinsProxySettings.getPassword();
+        Authenticator proxyAuthenticator = new Authenticator() {
+            @Override public Request authenticate(Route route, Response response) throws IOException {
+                String credential = Credentials.basic(username, password);
+                return response.request().newBuilder()
+                        .header("Proxy-Authorization", credential)
+                        .build();
+            }
+        };
+        Proxy proxy = JenkinsProxySettings.getJenkinsProxy()!=null ?  JenkinsProxySettings.getJenkinsProxy() : Proxy.NO_PROXY;
```

Comment:
We are building the authenticator above with the assumption that we have a proxy which will require auth as well.
Probably we can re-arrange the code in here a bit.
We can check if proxy exists at all or not... if no, no point in building the authenticator etc.

From a readability point of view...

Reply:
Done